atlp-rwanda/space-bn-backend
# Ft signup

### What does this PR do?
- Implement User registration and once Users are registered via the registration endpoint, should receive a JWT to be required on all subsequent calls to all other endpoints that require authentication.
### Description of Task to be completed?
- Implement  feature using swagger
- Implement JWT strategy for authentication.
- Implement password encryption
- Validation of inputs
- Write tests for user registration
### How should this be manually tested?
- Clone the repository.
- In your home directory  move  into the cloned folder by **cd ft-signup**
- Then run **npm install** to install all dependencies.
- Run **npm start** to start the server.
- after successful login u use "check-auth.js" which is inside src/middlewares to protect routes with the jwt token
- access the swagger documentation on: **localhost:5000/api-docs**
- Run  **npm run test** to run the test.
### Any background context you want to provide?
- N/A
### What are the relevant Pivotal Tracker stories?
- [#175743313](https://www.pivotaltracker.com/story/show/175743313)
- [#175743315](https://www.pivotaltracker.com/story/show/175743315)
### Screenshots (if appropriate)
- N/A
### Questions
- N/A
